src/rpdk/core/contract/resource_client.py

```diff
@@ -337,9 +349,16 @@ def call(self, action, request):
 
         while status == OperationStatus.IN_PROGRESS:
             callback_delay_seconds = self.assert_in_progress(status, response)
+            _status, read_response, _error = self.call_and_assert(
+                Action.READ, OperationStatus.SUCCESS, response["resourceModel"]
+            )
+            assert read_response["resourceModel"] == response["resourceModel"]
```

Comment:
I'm worried this may fail transiently if an update/create call is stabilizing. In between when the update/create is called and the read is called, a property could stabilize though it should happen very rarely. What is the reason to check for this?

Reply:
This was done to check if the handlers are returning the whole resource model as entered by the customer or just returning the resource model which has been updated. As mentioned in the contract test, "**A Create handler SHOULD return a model containing all properties set so far and nothing more during each IN_PROGRESS event.**"

Reply:
SHOULD != MUST, and therefore we cannot enforce this in contract test

It was written as SHOULD because this may be infeasible for many handlers

Reply:
makes sense. I will go ahead and remove this.